controllers/kubeadmconfig_controller.go

```diff
@@ -256,13 +256,21 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 	}
 
 	// Every other case it's a join scenario
-	// Nb. in this case ClusterConfiguration and JoinConfiguration should not be defined by users, but in case of misconfigurations, CABPK simply ignore them
+
+	//In case of join, ClusterConfiguration and InitConfiguration should not be defined by users
+	if config.Spec.ClusterConfiguration != nil || config.Spec.InitConfiguration != nil {
+		return ctrl.Result{}, errors.New("Control plane already exists for the cluster, KubeadmConfig objects with ClusterConfiguration or InitConfiguration can not be processed at this stage")
```

Comment:
It might be good to also emit an event here as well.

Reply:
If we return an error here, it means the KubeadmConfig will be reconciled again using backoff. We probably don't want to return an error.

Reply:
I asked @fabriziopandini to not make these changes as they are controversial and reduce this PR to fix the original issue of #213